internal/ip/ip_linux.go

```diff
@@ -22,13 +22,20 @@ package ip
 
 import (
 	"net"
+	"syscall"
+	"unsafe"
```

Comment:
in general, minimize `"unsafe"` code to the absolute minimum. in this case, that means splitting this file into `iplayer_linux.go` that contains the non-`unsafe`, linux-specific code.